packages/compass-import-export/src/export/export-json.ts

```diff
@@ -0,0 +1,129 @@
+import { Transform } from 'stream';
+import { pipeline } from 'stream/promises';
+import { EJSON } from 'bson';
+import type { Writable } from 'stream';
+import toNS from 'mongodb-ns';
+import type { DataService } from 'mongodb-data-service';
+
+import { createReadableCollectionCursor } from '../utils/collection-stream';
+import type {
+  ExportQuery,
+  ExportAggregation,
+} from '../utils/collection-stream';
+import { createDebug } from '../utils/logger';
+
+const debug = createDebug('export-json');
+
+type ExportJSONOptions = {
+  dataService: DataService;
+  ns: string;
+  output: Writable;
+  abortSignal: AbortSignal;
+  query?: ExportQuery;
+  aggregation?: ExportAggregation;
+  progressCallback: (index: number) => void;
+  variant: 'default' | 'relaxed' | 'canonical';
+};
+
+type ExportJSONResult = {
+  docsWritten: number;
+  aborted: boolean;
+};
+
+function getEJSONOptionsForVariant(
+  variant: 'default' | 'relaxed' | 'canonical'
+) {
+  if (variant === 'relaxed') {
+    return {
+      relaxed: true,
+    };
+  }
+  return variant === 'canonical'
+    ? {
+        relaxed: false, // canonical
+      }
+    : undefined; // default
+}
+
+export async function exportJSON({
+  dataService,
+  ns,
+  output,
+  abortSignal,
+  query = { filter: {} },
```

Comment:
We have to be able to export an aggregation as well, I think. So I'm wondering if we shouldn't just pass in something like a cursor. ie. build the whole query and run it on the outside. 🤔

Reply:
Good thinking. Updated the `exportJSON` function to take an `input` which is a cursor, then created an `exportJSONFromQuery` and `exportJSONFromAggregation`. Makes it a bit more general and makes these types nicer. I'm thinking when we integrate these we can revisit and see if there's a nicer interface to provide here 👍